src/runtime_src/driver/xclng/drm/xocl/subdev/mb_scheduler.c

```diff
@@ -2459,46 +2445,31 @@ static void destroy_client(struct platform_device *pdev, void **priv)
 		outstanding = new;
 	}
 
-	DRM_INFO("client exits pid(%d)\n", pid);
 
-	mutex_lock(&xdev->ctx_list_lock);
+	/* Unlock xclbin */
+	if (CLIENT_NUM_CU_CTX(client) != 0)
+		(void) xocl_icap_unlock_bitstream(xdev, xclbin_id, pid);
+
+	client->virt_cu_ref = 0;
+	client_release_implicit_cus(exec, client);
+	bitmap_zero(client->cu_bitmap, MAX_CUS);
+
 	list_del(&client->link);
-	mutex_unlock(&xdev->ctx_list_lock);
+	DRM_INFO("client exits pid(%d)\n", pid);
+
+	mutex_unlock(&xdev->dev_lock);
 
-	mutex_destroy(&client->lock);
 	devm_kfree(&pdev->dev, client);
 	*priv = NULL;
 }
 
 static uint poll_client(struct platform_device *pdev, struct file *filp,
 	poll_table *wait, void *priv)
 {
-	struct client_ctx	*client = (struct client_ctx *)priv;
-	struct exec_core	*exec;
-	int			counter;
-	uint			ret = 0;
-
-	exec = platform_get_drvdata(pdev);
+	struct exec_core *exec = platform_get_drvdata(pdev);
 
 	poll_wait(filp, &exec->poll_wait_queue, wait);
```

Comment:
What happens on time out?  The client will be woken up with POLLIN, but that's a change in behavior.

Reply:
Hmm.... it does not work, then. I'll have to think about it.